### PR TITLE
Modify QSPI example code so that it will work with nRF52x platforms

### PR DESCRIPTION
**Pull request by j3hill**
### Description

For the Nordic nRF52x platform, the QSPI flash memory address and the total flash buffer size need to be DWORD/4-byte aligned. This PR updates the example code to work with the Nordic nRF52840.

### Pull request type
    [X] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### CODE REVIEW

**Comment by j3hill** [comment]
@deepikabhavnani Could you please review this?

File: main.cpp
Comment (deepikabhavnani) [inline main.cpp]: Will `CMD_FASTREAD` work for all targets?

File: main.cpp
Comment (deepikabhavnani) [inline main.cpp]:
```suggestion
#if TARGET_NORDIC  // target specific check 
#define CMD_READ       0x0B     // Fast Read
#else
#define CMD_READ           0x03
#endif
```

Comment (j3hill) [reply main.cpp]: Fast read (0x0B) is more common with newer platforms/flash parts, especially ones that support QSPI. With that said, it makes sense to check the target and set the read command based on that. Thank you for the suggestion.

**Comment by j3hill** [comment]
@deepikabhavnani I've made the requested change.

**Review by deepikabhavnani — APPROVED** [review]
Looks good to me 👍

**Comment by deepikabhavnani** [comment]
@ARMmbed/mbed-os-storage - Please review.

@0xc0170 - Not sure if storage team has write access to this repo to perform merge, please see if you can give them access

**Review by offirko — APPROVED** [review]
LGTM

**Comment by offirko** [comment]
+ @maciejbocianski

@0xc0170 - as @deepikabhavnani wrote, I believe HAL ppl should give the final approval here.

File: main.cpp
Comment (yossi2le) [inline main.cpp:27]: I have a question regarding the TARGET_NORDIC. is all Nordic target should use the same fast read command?
I ask this because this will apply the change to all current and future Nordic boards and not only for nRF52x boards.
if you want to make this change only for the nRF52x boards you should use TARGET_NRF5x macro instead.

**Comment by 0xc0170** [comment]
I can merge this once @yossi2le approves. 

Looking at this repo, it has been idle for some time. needs readme, license :-) Do we still need this repository ? IF yes, will need an update and then adding to the official repos.

**Comment by yossi2le** [comment]
I would be happy to approve it but my question wasn't answered.

**Comment by deepikabhavnani** [comment]
> Do we still need this repository ?

Yes, this was official example for OOB testing of QSPI

Thanks for bringing this up, initially I thought this repo was for docs but docs uses this 
https://github.com/ARMmbed/mbed-os-examples-docs_only/blob/master/QSPI/main.cpp

@j3hill  - Can we update this.

**Comment by j3hill** [comment]
@yossi2le The nRF52840 is the first Nordic target to support QSPI functionality with external flash modules, and it requires fast reads. It is anticipated that future Nordic platforms (...and even faster external flash modules) will also require fast read support. 
@deepikabhavnani Yes. We can update the Mbed OS docs repository as well.

**Comment by yossi2le** [comment]
@j3hill thanks for your replay.

**Review by yossi2le — APPROVED** [review]
While this is a QSPI example and with the assumption that any future Nordic board with QSPI support will need this change then it is OK with me :)

**Pull request merged**
